controller/controller_server/csi_controller_server.py

```diff
@@ -400,9 +398,44 @@ def ControllerUnpublishVolume(self, request, context):
 
     def ValidateVolumeCapabilities(self, request, context):
         logger.info("ValidateVolumeCapabilities")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished ValidateVolumeCapabilities")
-        return csi_pb2.ValidateVolumeCapabilitiesResponse()
+        try:
+            utils.validate_validate_volume_capabilities_request(request)
+
+            secrets = request.secrets
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+
```

Comment:
IMO, no need for this blank line, and same for those inside the `with get_agent` block

Reply:
done

Reply:
**same for those inside the with get_agent block**

Reply:
I dont think it right in `get_agent`. the validation and the get object are independent

Reply:
what about the blank line between the `with` and the validation?

Reply:
ok, done